include/grass/imagery.h

```diff
@@ -51,20 +51,22 @@ struct Control_Points
 
 struct One_Sig
 {
-    char desc[100];     /* name of target class */
+    char desc[256];     /* name of target class */
     int npoints;        /* cell count used to determine class parameters */
     double *mean;		/* one mean for each band */
     double **var;		/* covariance band-band   */
     int status;		/* may be used to 'delete' a signature */
     float r, g, b;		/* color */
     int have_color;
+    int oclass;         /* original class value */
 };
 
 struct Signature
 {
     int nbands;         /* band (imagery group member) count */
     char **semantic_labels;    /* list of semantic labels */
     int nsigs;          /* signature count */
+    int have_oclass;     /* is One_Sig.oclass filled with values? */
```

Comment:
Booleans should be `bool` at this point.

Reply:
So be it. Only now it will rise a question why other bool properties in the struct are int but this one is bool.

Reply:
I think we shifted from purely preferring consistency to adding modern code even when surrounding code is in legacy style. At least I did. :-)

Reply:
Boolean in this case is not worth the trouble – there is no printf/scanf formatting type for bool. Not an issue for printf as it promotes anything smaller than int to int, but scanf is more problematic as an intermediate variable has to be used to read into int and then cast back to bool.

Reply:
I wouldn't say it is a showstopper in general, but it does weight against using bool here.

Reply:
Sorry, I'm too lazy to work around bool as int works just fine & matches the code style.